scripts/ec2cleanup.py

```diff
@@ -129,7 +133,35 @@ def delete_unused_network_interfaces(provider_mgmt, excluded_enis, output):
         logger.exception('Exception in %r', delete_unused_network_interfaces.__name__)
 
 
-def ec2cleanup(exclude_volumes, exclude_eips, exclude_elbs, exclude_enis, output):
+def delete_stacks(provider_mgmt, excluded_stacks, stack_template, output):
+    stack_list = []
+    provider_name = provider_mgmt.kwargs['name']
+    try:
+        for stack in provider_mgmt.list_stack():
+            if (excluded_stacks and
+                    stack.stack_name in excluded_stacks or
+                    not stack.stack_name.startswith(stack_template)):
+                logger.info("  Excluding Stack name: %r", stack.stack_name)
+                continue
+            else:
+                today = datetime.utcnow().replace(tzinfo=None)
+                some_date = today - timedelta(days=1)
+                if stack.creation_time < some_date:
+                    stack_list.append([provider_name, stack.stack_name])
+                    provider_mgmt.delete_stack(stack.stack_name)
```

Comment:
REQUIRED: if  some exception happens during stack deletion, the rest of stacks won't be removed. 
I believe you need to wrap try/catch around delete_stack operation instead of whole method.